src/view.cpp

```diff
@@ -368,22 +368,25 @@ void View::open_in_pager(const std::string& filename)
 }
 
 nonstd::optional<std::uint8_t> View::open_in_browser(const std::string& url,
-	const std::string& feedurl, const std::string& type, bool interactive)
+	const std::string& feedurl, const std::string& type, const std::string& title, bool interactive)
 {
 	std::string cmdline;
 	const std::string browser = cfg->get_configvalue("browser");
 	const std::string escaped_url = "'" + utils::replace_all(url, "'", "%27") + "'";
 	const std::string escaped_feedurl = "'" + utils::replace_all(feedurl, "'",
 			"%27") + "'";
 	const std::string quoted_type = "'" + type + "'";
+	const std::string quoted_title = "'" + title + "'";
```

Comment:
The title might contain quotes, so you should invoke `utils::replace_all` to change them to `%27` (see a few lines above). `type` doesn't need this treatment because it can only take a limited number of values that come from inside Newsboat, and none of them contain quotes.

Reply:
Sorry I missed that, I fixed it in the latest commit  #([62e792b](https://github.com/newsboat/newsboat/pull/2261/commits/62e792bc4d0505b76c37fbc3333d1dd3fcaf8b68)). I've also changed the variable name (to `escaped_title`) to align with the previous variables. Thank you for the feedback :)

Reply:
Apologies, but I just realized that `replace_all` should replace quotes in title with something like `'\''`, not `%27`. The two calls above are for URLs, which use percent-encoding, so it makes sense to replace with a percent-encoded quote there, which is `%27`. However, the title is not a URL, it's a string, and it's not percent-encoded — that's why we should handle it differently.

Sorry to turn your first contribution into such an onerous experience, but could you please change the code again, and test that it works with titles that contain quotes? (You could set `browser` to a script that just puts its argument into a file, and examine that.)

Reply:
No worries, it isn't any trouble at all, this is quite a valuable learning experience, so thank you!

I found that only replacing `'` with `\'` returns an error in the script for titles with single quotes, so I replaced all the `"` with `\"` and `'` with `\'` while surrounding the entire title with `"`:
```
const std::string escaped_title = '"' + utils::replace_all(title, std::vector<std::pair<std::string, std::string>> {{"\"", "\\\""}, {"'", "\'"}}) + '"';
```
I didn't commit it yet, I wanted to get your opinion on this change to make sure it is reasonable. I tested the above solution with multiple titles: with one `'`, two `'`, and two `"`, and it works well (at least with the script shown below). For e.g. the title `"Cuca \"Italiana\""` is changed to `\"Cuca \\\"Italiana\\\"\"` which is passed to the script (shown below).

The script I used in place of the `browser` (I am not very good with bash scripting, so this script might be the reason it breaks for titles with a single quote, if this is the case then your proposed solution should work):
```
#!/usr/bin/env bash

for arg in "$@"; do
    echo "${arg}" >> "/home/aneesh/test.out"
done
```
If this change sounds correct, let me know so I can commit it :)

Reply:
Sorry I'm so slow to respond!

Enclosing the title in double quotes is problematic, because they allow much more stuff in them, including variables and even command invocations. A title like `How I organize my $HOME` will turn into `How I organize my /home/minoru` (on my machine) when passed to the script. See §2.2 in https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html. That's just asking to get pwned, so I'd prefer not to go down that path.

The proper way to handle this is to call `execv` and pass the arguments as separate elements of `argv`, but that requires way more changes (including the Rust side and Rust-C++ bridge). I'd like to get there one day, but it's too big an ask for this PR.

That leaves us with somewhat hacky solutions, e.g. turn `it's` into `'it'\''s'`. I think this can be accomplished with the help of one of the `tokenize` methods from the utils.hpp file. Wanna try?

Reply:
I apologise for my lack of patience,

Here is my attempt at the fix:

```
std::string escaped_title = "";
std::vector<std::string> tokenized_title = utils::tokenize_spaced(title, "'");
for (std::string title_chunk : tokenized_title) {
    if (title_chunk == "'") {
        escaped_title += "\\\'";
    } else {
        escaped_title += "'" + title_chunk + "'";
    }
}
```
Would it be better for this to be in another function, perhaps named `preserve_quotes` in `utils.cpp`?
Something like this:
```
std::string utils::preserve_quotes(const std::string &s)
{
     std::string quoted_string = "";
     std::vector<std::string> string_tokenized = tokenize_spaced(s, "'");
     for (std::string title_chunk : string_tokenized) {
         if (title_chunk == "'") {
             quoted_string += "\\\'";
         } else {
             quoted_string += "'" + title_chunk + "'";
         }
     }

     return quoted_string;
}
```

Reply:
Yeah, this code seems right. If you pull it out into a function, please add a doc-comment briefly explaining what it does, and add some tests in _tests/utils.cpp_.